lib/stats/stats-registry.c

```diff
@@ -140,8 +140,12 @@ _register_counter(gint stats_level, const StatsClusterKey *sc_key, gint type,
   sc = _grab_cluster(stats_level, sc_key, dynamic);
   if (sc)
     {
+      StatsCounterItem *ctr = stats_cluster_get_counter(sc, type);
```

Comment:
These 4 lines confuse me. What is the goal here? To support registering already registered counters? It would be cleaner if the stats counters need to be unregistered first.

`stats_cluster_get_counter` can only return tracked counters. So the `stats_cluster_track_counter` is not needed if `stats_cluster_get_counter` returns with a counter. Not necessarily wrong (due that bit-or is idempotent), but confusing. Maybe it is called at a wrong place?

Reply:
`track_counter` is different: it also increases the `use_count`.

"Currently, it is possible to register a `StatsCounterItem` and get back an external counter.":
These 4 lines implements this.

Reply:
I understand now, thanks.